src/sources/soundsourceffmpeg.cpp

```diff
@@ -1,22 +1,27 @@
 #include "sources/soundsourceffmpeg.h"
 
+#include "encoder/encoderffmpegresample.h"
+
 #include <mutex>
 #include <vector>
 
 #define AUDIOSOURCEFFMPEG_CACHESIZE 1000
-#define AUDIOSOURCEFFMPEG_MIXXXFRAME_TO_BYTEOFFSET (sizeof(CSAMPLE) * getChannelCount())
+#define AUDIOSOURCEFFMPEG_MIXXXFRAME_TO_BYTEOFFSET(numFrames) (frames2samples(numFrames) * sizeof(CSAMPLE))
```

Comment:
I would prefer these to be functions instead of macros but I don't insist for this PR since these cleanups are great on their own :).